dbms/src/Flash/Mpp/ExchangeReceiver.cpp

```diff
@@ -347,15 +350,18 @@ ExchangeReceiverBase<RPCContext>::ExchangeReceiverBase(
 template <typename RPCContext>
 ExchangeReceiverBase<RPCContext>::~ExchangeReceiverBase()
 {
+    bool wait_all_connection_done = false;
     try
     {
         close();
         waitAllConnectionDone();
+        wait_all_connection_done = true;
         thread_manager->wait();
         ExchangeReceiverMetric::clearDataSizeMetric(data_size_in_queue);
     }
     catch (...)
     {
+        RUNTIME_ASSERT(wait_all_connection_done, "We should wait the close of all connections");
```

Comment:
So you prefer `assert(wait_all_connection_done)` over `assert(live_connections == 0)` here?

Reply:
> So you prefer `assert(wait_all_connection_done)` over `assert(live_connections == 0)` here?

I forget to change this.